dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/XPathMapper.java

```diff
@@ -70,13 +71,101 @@
         rootPaths.toArray(new String[rootPaths.size()]), defaultRootNamespace));
   }
 
-  private XPathMapper(Document doc, String[] rootPaths, String defaultRootNamespace) {
+  public XPathMapper(Document doc, String[] rootPaths, String defaultRootNamespace) {
     this.xpw = new XPathWrapper(doc);
     this.rootPaths = new HashSet<>(Arrays.asList(rootPaths));
     this.defaultRootNamespace = defaultRootNamespace;
   }
 
-  public Set<String> getVariables(String templateString) {
+  /**
+   * The central entry method to set up an XPathMapper
+   * @param doc the document for evaluation
+   * @param bindMapperClass the bindMapper class
+   * @param rootPaths optional array of root paths
+   * @return The bind mapper with filled setters and fields, ready for getter consumption
+   */
+  public static <T> T readDocument(Document doc, Class<T> bindMapperClass, String... rootPaths) {
+    try {
+      XPathRoot rootAnnotation = bindMapperClass.getAnnotation(XPathRoot.class);
+      T bindMapper = bindMapperClass.getDeclaredConstructor().newInstance();
+
+      ensureValidRootPaths(rootAnnotation, rootPaths);
+
+      XPathMapper xPathMapper = new XPathMapper(doc, determineRootPaths(rootAnnotation, rootPaths),
+          determineDefaultRootNamespace(rootAnnotation));
+
+      List<Method> methodsWithAnnotation = ReflectionUtils.getMethodsAnnotatedWith(bindMapperClass, XPathBinding.class);
+      for (Method m : methodsWithAnnotation) {
+        ReflectionUtils.ensureAccess(m);
+        m.invoke(bindMapper, xPathMapper.invoke(bindMapper, m, null));
```

Comment:
`invoke` bitte unbedingt umbenennen in irgendwas wie `determineValueForField` oder so, `invoke` war vorgegeben durch die Proxy API, wir können jetzt sprechender sein.

Reply:
Die beiden `invoke` sind aber fest vorgegeben, einmal von `Method` und einmal durch das `@Override` im `XPathMapper`. 

Ich könnte höchstens das `invoke` des `XPathMapper` an eine Methode `determineValueForField` delegiern lassen.

Reply:
Ah, sorry, gestern zu früh kommentiert hier. Das hängt zusammen mit meinem Kommentar bzgl des Entferrnes des Proxy-Ansatzes. Wenn der weg ist kann man das auch sprechender benennen.